private/createConstraintGradientFunction.m

```diff
@@ -22,21 +23,45 @@ function createConstraintGradientFunction(N,useMaxNorm)
 c4 = g(:,2)'*g(:,2)-integralConstraint;
 c5 = g(:,3)'*g(:,3)-integralConstraint;
 
-if useMaxNorm == false  
-    c2 = (sum(g.^2,2)-gMax^2)';%Nonlinear inequality constraint <= 0   
+x = 1;
+y = 2;
+z = 3;
+
+% Constraint for compensation of Maxwell terms (concomitant fields)
+% For theory, please read/cite the following abstract (or later paper):
+% Filip Szczepankiewicz and Markus Nilsson
+% "Maxwell-compensated waveform design for asymmetric diffusion encoding"
+% ISMRM 2018, Paris, France
+% Download PDF at: https://goo.gl/vVGQq2
+
+gxx = sum( g(:,x).*g(:,x).*s_vec );
+gyy = sum( g(:,y).*g(:,y).*s_vec );
+gzz = sum( g(:,z).*g(:,z).*s_vec );
+gxy = sum( g(:,x).*g(:,y).*s_vec );
+gxz = sum( g(:,x).*g(:,z).*s_vec );
+gyz = sum( g(:,y).*g(:,z).*s_vec );
```

Comment:
Consider if we should use the integration matrix (trapezoid rule) or use it as dot product, as above, instead.

Reply:
What ever works best. I consider my code as a quick-and-dirty test. I am sorry for any crappy solutions.

Reply:
The dot product is the correct implementation. Rewrote using matrix algebra.